src/sentry/incidents/utils/sentry_apps.py

```diff
@@ -2,21 +2,32 @@
 
 from rest_framework import serializers
 
+from sentry.auth.access import SystemAccess
 from sentry.incidents.logic import get_filtered_actions
 from sentry.incidents.models import AlertRuleTriggerAction
+from sentry.incidents.serializers import AlertRuleTriggerActionSerializer
 from sentry.mediators import alert_rule_actions
 from sentry.models import SentryAppInstallation
 
 
-def trigger_sentry_app_action_creators_for_incidents(
-    validated_alert_rule_data: Mapping[str, Any]
-) -> None:
+def trigger_sentry_app_action_creators_for_incidents(alert_rule_data: Mapping[str, Any]) -> None:
 
     sentry_app_actions = get_filtered_actions(
-        validated_alert_rule_data=validated_alert_rule_data,
+        alert_rule_data=alert_rule_data,
         action_type=AlertRuleTriggerAction.Type.SENTRY_APP,
     )
-    for action in sentry_app_actions:
+    # We're doing this so that Sentry Apps without alert-rule-action schemas still get saved
+    sentry_app_actions_with_components = list(
```

Comment:
You should consider adding this filter to `get_filtered_actions()` instead of iterating twice.
```py
return [
    rewrite_trigger_action_fields(action)
    for trigger in alert_rule_data.get("triggers", [])
    for action in trigger.get("actions", [])
    if (
        STRING_TO_ACTION_TYPE.get(action.get("type")) == action_type
        # We're doing this so that Sentry Apps without alert-rule-action schemas still get saved
        and (not with_components or action.get("sentry_app_config"))
    )
]
```

Reply:
@mgaeta  I considered this, but I eventually want to use `get_filtered_actions` for `get_slack_channel_ids` in `src/sentry/incidents/logic.py`, so I'd prefer to keep it agnostic to the type it is filtering by.

Reply:
Ok don't actually do this but I thought you'd like to see another pattern you could use:
```py
def get_filtered_actions(
    alert_rule_data: Mapping[str, Any],
    action_type: AlertRuleTriggerAction.Type,
    filters: Sequence[Callable[[Sequence[Mapping[str, Any]]], bool]]
) -> Sequence[Mapping[str, Any]]:
    from sentry.incidents.serializers import STRING_TO_ACTION_TYPE

    return [
        rewrite_trigger_action_fields(action)
        for trigger in alert_rule_data.get("triggers", [])
        for action in trigger.get("actions", [])
        if (
            STRING_TO_ACTION_TYPE.get(action.get("type")) == action_type
            and all(f(action) for f in filters)
        )
    ]
```